pilot/pkg/bootstrap/server.go

```diff
@@ -229,6 +229,12 @@ func NewServer(args *PilotArgs, initFuncs ...func(*Server)) (*Server, error) {
 	// used for both initKubeRegistry and initClusterRegistries
 	args.RegistryOptions.KubeOptions.EndpointMode = kubecontroller.DetectEndpointMode(s.kubeClient)
 
+	enableEndpointSlice, _ := features.EnableEndpointSliceController()
+	if features.EnableDualStack && !enableEndpointSlice {
```

Comment:
For regular endpoints, only the default (or first IP family type based on the Service) is reported, then prefer `PILOT_USE_ENDPOINT_SLICE` to retrive the endpoint's IPv4 and IPv6 addresses.

Reply:
Since this doesn't take DetectEndpointMode into account it seems inaccurate? Or why not just set PILOT_USE_ENDPOINT_SLICE to true if DUAL_STACK is set if its required. Having this coupling seems to just make usage more complex.

Reply:
Q: Since this doesn't take DetectEndpointMode into account it seems inaccurate? 
A: Good idea,  I will use `DetectEndpointMode` to check the endpoint mode which is more accurate.